react/SummaryInstallments.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import { defineMessages, FormattedNumber } from 'react-intl'
+import { Installment, InstallmentOption } from 'vtex.checkout-graphql'
+import { useCssHandles } from 'vtex.css-handles'
+import { FormattedCurrency } from 'vtex.format-currency'
+import { IOMessageWithMarkers } from 'vtex.native-types'
+import { Loading } from 'vtex.render-runtime'
+
+import { useSummary } from './SummaryContext'
+
+const CSS_HANDLES = [
+  'installments',
+  'installmentsNumber',
+  'installmentValue',
+  'installmentsTotalValue',
+  'interestRate',
+  'paymentSystemName',
+] as const
+
+function SummaryInstallments(props: SummaryInstallmentsProps) {
+  const { loading, paymentData } = useSummary()
+  const { message, markers } = props
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const { installmentOptions } = paymentData
+
+  if (loading) {
+    return <Loading />
+  }
+
+  if (!installmentOptions) {
+    return null
+  }
+
+  let maxInstallments: InstallmentOption | undefined
+
+  installmentOptions.forEach((installmentOption) => {
+    const currentValueIsEmpty =
+      !maxInstallments || Object.keys(maxInstallments).length === 0
+
+    if (
+      currentValueIsEmpty ||
+      installmentOption.installments.length >
+        (maxInstallments?.installments.length ?? 0)
+    ) {
+      maxInstallments = installmentOption
+    }
+  })
+
+  const maxInstallment: Installment | undefined =
+    maxInstallments?.installments[maxInstallments?.installments.length - 1]
+
+  const installmentsNumber = maxInstallment?.count
+  const installmentValue = maxInstallment?.value
+  const installmentsTotalValue = maxInstallment?.total
+  const interestRate = maxInstallment?.interestRate
+  const paymentSystemName = maxInstallments?.paymentSystem
+  const hasInterest = maxInstallment?.hasInterestRate
```

Comment:
Have you ever tried to use destructor and rename the properties?

Reply:
Adjusted